sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -1198,8 +1099,8 @@ protected void writeFooter() throws Exception {
       }
 
       @Override
-      public void write(T value) throws Exception {
-        coder.encode(value, out, Context.OUTER);
+      public void write(String value) throws Exception {
+        DEFAULT_TEXT_CODER.encode(value, out, Context.OUTER);
```

Comment:
Now that we know we're only writing utf8 string, we can simplify this too. Replace `this.out` with an `OutputStreamWriter` in `utf8` mode, then just use  `osw.write(value)`.

http://stackoverflow.com/questions/37889354/best-way-to-write-a-string-to-outputstream-in-java